src/core/src/sys/cache.rs

```diff
@@ -162,21 +162,6 @@ impl EnergyCache {
             }
         }
 
-        // Interactions withing the sub-system being moved
-        for (i, &part_i) in idxes.iter().enumerate() {
-            for (j, &part_j) in idxes.iter().enumerate().skip(i + 1) {
-                let r = system.cell().distance(&newpos[i], &newpos[j]);
-                let energy = evaluator.pair(r, part_i, part_j);
-
-                pairs_delta += energy;
-                new_pairs[(part_i, part_j)] += energy;
-                new_pairs[(part_j, part_i)] += energy;
-
-                pairs_delta -= self.pairs_cache[(part_i, part_j)];
-            }
-        }
-
```

Comment:
I think I wrote this code to handle the case of flexible molecules. As it is tested (removing it broke the tests), we can just keep it around.